### PR TITLE
url part of footnotes and links shall not be checked

### LINKED ISSUE #97: url part of footnotes and links shall not be checked
**Describe the bug**

Links are also tokenized and spellchecked, which leads to false positives

```
error: spellcheck(Hunspell)
   --> /tmp/build/83e750f6/git-spearow-juice/coaster/src/frameworks/opencl/program.rs:10
    |
 10 |  [binary]: ../../binary/index.html
    |                               ^^^^
    | - HTML
    |
    |   Possible spelling mistake found.
```

with a source snippet

```rust
#[derive(Debug, Copy, Clone)]
/// Defines a OpenCL Program.
///
/// A Program is OpenCL's version of Coaster's [binary][binary].
/// [binary]: ../../binary/index.html
pub struct Program {
    id: isize,
}
```

**To Reproduce**

Steps to reproduce the behaviour:

1. Run 'cargo spellcheck -vvv' on a doc comment that contains a footnote / link
2. See error

**Expected behavior**

Ignore the url part of the link index, the `title` of the link is already checked of the inline test but for sake of interactive convenience and not breaking the structure, it should be suggested in both cases.

**Please complete the following information:**
 - System: Fedora 32 x86_64
 - Obtained: git
 - Version: `ab2819f595872ea12d52b823f7d09a78c91085eb`

**Additional context**
https://ci.spearow.io/teams/main/pipelines/cargo-spellcheck/jobs/pr-validate/builds/282

### PR DESCRIPTION
**Pull request by drahnr**
## What does this PR accomplish?

 * 🩹 Bug Fix

@laysauchoa  you should be able to push to this branch directly

Ref #102 .
Closes #97.

## 📜 Checklist

 * [x] Works on the `./demo` sub directory
 * [x] Test coverage is excellent and passes
 * [x] Documentation is thorough

### CODE REVIEW

File: src/traverse/mod.rs [hunk 1]
```diff
@@ -350,7 +350,7 @@ pub(crate) fn extract(
                 if cargo_toml.is_file() {
                     Extraction::Manifest(cargo_toml)
                 } else {
-                    // @todo should we just collect all .rs files here instead?
+                    // TODO should we just collect all .rs files here instead?
```
Comment (laysauchoa) [inline src/traverse/mod.rs:353]: out of curiousity, why use `TODO` instead of `@todo`?

Comment (drahnr) [reply src/traverse/mod.rs:353]: It's more common generally, and @ is kind of associated to mentions.
Also there is a nice vs code plugin which adds a pane for all todos and fixmes in a codebase.

Comment (drahnr) [reply src/traverse/mod.rs:353]: The plugin is called `Todo Tree`

**Comment by drahnr** [comment]
Once this is merged there will be a `v0.4.4` release

**Pull request merged**